File: Army(java)/Warriors/Soldier.java
package Warriors;

import EnumID.*;
import Basic.*;

public class Soldier extends Unit {
    private WarriorID warrior;
    
    public Soldier(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.SOLDIER;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }
}

File: Army(c++)/Warriors/Soldier.cpp
#include "Soldier.hpp"

Soldier::Soldier(const std::string& name, int hp, int dmg) 
    : Unit(name, hp, dmg, new SoldierState(this))
    , warrior(WarriorID::SOLDIER) {}

Soldier::~Soldier() {}

WarriorID Soldier::getWarriorID() const {
    return this->warrior;
}

void Soldier::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}
